src/model.ts

```diff
@@ -291,6 +291,27 @@ export class GitExtension implements IGitExtension {
     return this._dirtyStagedFilesStatusChanged;
   }
 
+  /**
+   * Boolean indicating whether credentials are required from the user.
+   */
+  get credentialsRequired(): boolean {
+    return this._credentialsRequired;
+  }
+
+  set credentialsRequired(value: boolean) {
+    if (this._credentialsRequired !== value) {
+      this._credentialsRequired = value;
+      this._credentialsRequiredSignal.emit(value);
+    }
+  }
+
+  /**
+   * A signal emitted whenever credentials are required, or are not required anymore.
+   */
+  get credentialsRequiredSignal(): ISignal<IGitExtension, boolean> {
```

Comment:
Let's use a more Jupyter wording

```suggestion
  get credentialsRequiredChanged(): ISignal<IGitExtension, boolean> {
```